docs/components/mdx-components.js

```diff
@@ -27,9 +43,87 @@ const InlineCode = ({ children }) => (
   </code>
 );
 
+const Blockquote = ({ children, ...props }) => (
+  <blockquote {...props}>
+    {children}
+    <style jsx>
+      {`
+        blockquote {
+          color: #6a737d;
+          border-left: .25em solid #dfe2e5;
+          padding: 0 1em;
+          margin: 0 0 12px 0;
+        }
+    `}
+    </style>
+  </blockquote>
+);
+
+
+const hSpacing = { mb: '24px', mt: '16px' };
+const PermalinkHeading = ({ children, type, ...props }) => {
+  const slugger = new GithubSlugger();
+
+  let text = children || '';
+
+  // If there are sub components, convert them to text
+  if (Array.isArray(children)) {
+    text = children
+      .map((child) => (typeof child === 'object' ? child.props.children : child))
+      .join('');
+  }
+
+  const id = slugger.slug(text);
+
+  return (
+    <Heading id={id} {...hSpacing} type={type} {...props}>
+      <a href={`#${id}`}>
+        {children}
+      </a>
+      <span className="permalink"><PermalinkIcon /></span>
+
+      <style jsx>
+        {`
+          a {
+            text-decoration: none;
+            color: inherit;
+            cursor: pointer;
+          }
+          .permalink {
+            visibility: hidden;
+            text-align: center;
+            vertical-align: middle;
+
+            float: left;
+            padding-right: 4px;
+            margin-left: -20px;
+            line-height: 1;
+          }
+          a:hover + .permalink {
+            visibility: visible;
+          }
+          a:hover {
+            text-decoration: underline;
+          }
+      `}
+
+      </style>
+    </Heading>
+  );
+};
+
 const components = {
-  h1: (props) => <Heading type="h1" {...props} />,
-  h2: (props) => <Heading type="h2" {...props} />,
+  p: (props) => <Text as="p" type="body" {...props} />,
+  h1: (props) => <PermalinkHeading type="h1" {...props} />,
+  h2: (props) => <PermalinkHeading type="h2" {...props} />,
+  h3: (props) => <PermalinkHeading type="h3" {...props} />,
+  h4: (props) => <PermalinkHeading type="h4" {...props} />,
+  // don't have styling for h5 or h6, use h4
+  h5: (props) => <PermalinkHeading type="h1" {...props} />,
+  h6: (props) => <PermalinkHeading type="h1" {...props} />,
```

Comment:
is this supposed to be `type="h4"` or am i misunderstanding?

Reply:
haha yes ill fix that